### DISCUSSION TITLE
Improve graph display

### DISCUSSION THREAD

**Discussion by xplosionmind** [maximevaillancourt/digital-garden-jekyll-template#81] [Ideas]
Having a graph which displays the connection among notes is great, but as of right now it's kind of limited, and it doesn't feel very ergonomic. A few ideas of improvements may be:

- [ ] Showing the current note at the center of the graph
- [ ] Resizing the graph to make all the content fit
- [ ] Simulating “forces” and attractions (as in Obsidian's graph, for example)

## Inspiration

Some code for inspiration:
- <https://github.com/hugcis/hugcis.github.io/tree/master/js>

**Comment by Am3ra** [comment]
Would adding weight to notes with more connections be relevant?

**Comment by vlwkaos** [comment]
https://github.com/maximevaillancourt/digital-garden-jekyll-template/blob/ac70b2f668e06de596ad54395d9029766f535c32/_includes/notes_graph.html#L63

I am a bit lazy to make a PR. so I will leave it here. During node update, indegree check condition is never true because `l.source` is an object not an id. So the correct condition should be `l.source.id === el.id || l.target.id === el.id` to match the created json file.

**Comment by adrien-perello** [comment]
On the [Foam discord](https://discord.com/channels/729975036148056075/735778843151040512/850931487187402793), someone managed to leverage the Foam graph (see [here](https://hukacode.github.io/graph/) or [here](https://foam-template-gatsby-kb-hikerpig.vercel.app/) for an idea)

> I get the code in this file.
https://github.com/foambubble/foam/blob/master/packages/foam-vscode/static/dataviz/graph.js
>
> Add 2 libaries
> ```<script src="//unpkg.com/force-graph"></script>```
> ```<script src="https://d3js.org/d3.v6.min.js%22%3E</script>```
>
> Then write a script to build graphData.

Maybe that's an idea?

**Comment by florhizome** [comment]
To me the question is if I’d want one global graph holding a lot of information (tags, maybe a cumulated attribute like relativity, possible a distinct type of reference for references to book/literature notes) that adapts to the site where it’s displayed and can be manipulated by the visitor with checkboxes or so, or having multiple graphs generated for distinct usages (between a “global view” that gives a useful quick overview on what’s on the site and a detailed, zoomed in “local view”) . I would not know how that would impact performance though. I think for visualization to be really useful long term it would benefit from such adaptability.

**Comment by manunamz** [comment]
I recommend using the [`force-graph`](https://github.com/vasturiano/force-graph) APIs for this.

**Comment by chaosarium** [comment]
For inspiration, it looks like that the [quartz](https://github.com/jackyzha0/quartz) template provides a much smoother graph experience (check out https://quartz.jzhao.xyz for demo). It was done using around 200 lines of JS https://github.com/jackyzha0/quartz/blob/hugo/assets/js/graph.js.

**Comment by candideu** [comment]
Has anyone been able to figure this out? I'm looking for a more fluid graph, similar to the one featured on the [d3.js Disjoint force-directed graph demo](https://observablehq.com/@d3/disjoint-force-directed-graph/2) and in Obsidian + Logseq.

![](https://i.imgur.com/CaP8aXT.gif)

I would also love to have a page-specific graph like Obsidian/Logseq, which either only displays the other notes connected to that note (related: https://github.com/maximevaillancourt/digital-garden-jekyll-template/discussions/175):

![image](https://github.com/user-attachments/assets/c9ecb162-ee0b-48df-9dc1-6ae11a8fcecb)

OR displays the entire graph, but highlights its connected nodes, like what we currently see when hovering over the node:

![image](https://github.com/user-attachments/assets/41778a42-c147-4e8d-8fd6-05cda8e02729)
